File: baekjoon/단계별로 풀어보기/함수/1065.js
const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
})

let input = []

const isHansu = (n) => {
  const ns = `${n}`
  if (ns.length === 1) return true
  let dis = []
  for (let i = 0; i < ns.length - 1; i++) {
    dis.push(ns[i] - ns[i + 1])
  }
  if (new Set(dis).size === 1) return true
  return false
}
const handleInput = (line) => {
  input = Number(line)
}
const handleOutput = () => {
  let count = 0
  for (let i = 1; i <= input; i++) {
    if (isHansu(i)) {
      count++
    }
  }
  console.log(count)
}

rl.on('line', handleInput).on('close', handleOutput)

File: baekjoon/단계별로 풀어보기/함수/1065.py
def isHanSu(n):
  strN = str(n)
  if len(strN) == 1:
    return True
  else:
    chaSet = set()
    for i in range(len(strN) - 1):
      chaSet.add(int(strN[i]) - int(strN[i+1]))
    if len(list(chaSet)) == 1:
      return True
    else:
      return False

N = int(input())
count = 0
for i in range(N):
  if isHanSu(i+1):
    count = count + 1

print(count)